tests/models/patchtst/test_modeling_patchtst.py

```diff
@@ -0,0 +1,384 @@
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.mask_type = mask_type
+        self.random_mask_ratio = random_mask_ratio
+
+        self.seed = seed
+        self.num_targets = num_targets
+        self.distil = distil
+        self.num_patches = (max(self.context_length, self.patch_length) - self.patch_length) // self.patch_stride + 1
+
+    def get_config(self):
+        return PatchTSTConfig(
+            prediction_length=self.prediction_length,
+            patch_length=self.patch_length,
+            patch_stride=self.patch_stride,
+            num_input_channels=self.num_input_channels,
+            d_model=self.hidden_size,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            ffn_dim=self.intermediate_size,
+            dropout=self.hidden_dropout_prob,
+            attention_dropout=self.attention_probs_dropout_prob,
+            context_length=self.context_length,
+            activation_function=self.hidden_act,
+            seed=self.seed,
+            num_targets=self.num_targets,
+            mask_type=self.mask_type,
+            random_mask_ratio=self.random_mask_ratio,
+        )
+
+    def prepare_patchtst_inputs_dict(self, config):
+        _past_length = config.context_length
+        # bs, num_input_channels, num_patch, patch_len
+
+        # [bs x seq_len x num_input_channels]
+        past_values = floats_tensor([self.batch_size, _past_length, self.num_input_channels])
+
+        future_values = floats_tensor([self.batch_size, config.prediction_length, self.num_input_channels])
+
+        inputs_dict = {
+            "past_values": past_values,
+            "future_values": future_values,
+        }
+        return inputs_dict
+
+    def prepare_config_and_inputs(self):
+        config = self.get_config()
+        inputs_dict = self.prepare_patchtst_inputs_dict(config)
+        return config, inputs_dict
+
+    def prepare_config_and_inputs_for_common(self):
+        config, inputs_dict = self.prepare_config_and_inputs()
+        return config, inputs_dict
+
+
+@require_torch
+class PatchTSTModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
+    all_model_classes = (
+        (
+            PatchTSTModel,
+            PatchTSTForPrediction,
+            PatchTSTForPretraining,
+            PatchTSTForClassification,
+            PatchTSTForRegression,
+        )
+        if is_torch_available()
+        else ()
+    )
+    all_generative_model_classes = (
+        (PatchTSTForPrediction, PatchTSTForRegression, PatchTSTForPretraining) if is_torch_available() else ()
+    )
+    pipeline_model_mapping = {"feature-extraction": PatchTSTModel} if is_torch_available() else {}
+    test_pruning = False
+    test_head_masking = False
+    test_missing_keys = False
```

Comment:
Why is this set to `False` for this model?

Reply:
@amyeroberts  Are you referring to `test_head_masking ` being set to True? We don't have an option to set `head_mask` in the model forward calls, so we set it to False.

Reply:
Sorry - I wasn't clear here. I was referring to `test_missing_keys`

Reply:
@amyeroberts I see. I have set `test_missing_keys=True` and the test passes.